File: src/main/java/com/wafflestudio/draft/model/Device.java
package com.wafflestudio.draft.model;

import lombok.Getter;
import lombok.NoArgsConstructor;
import lombok.RequiredArgsConstructor;
import lombok.Setter;

import javax.persistence.*;
import javax.validation.constraints.NotBlank;


@Entity
@Getter
@Setter
@RequiredArgsConstructor
@NoArgsConstructor
public class Device extends BaseTimeEntity {
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;

    @NotBlank
    @Column(unique = true)
    private String deviceToken;

    @ManyToOne(fetch = FetchType.LAZY)
    @JoinColumn(name = "user_id", referencedColumnName = "id")
    private User user;

}

File: src/main/kotlin/com/wafflestudio/draft/model/Device.kt
package com.wafflestudio.draft.model

import javax.persistence.*
import javax.validation.constraints.NotBlank

@Entity
data class Device(
        @field:NotBlank
        @Column(unique = true)
        var deviceToken: String,

        @Id
        @GeneratedValue(strategy = GenerationType.IDENTITY)
        var id: Long? = null,

        @ManyToOne(fetch = FetchType.LAZY)
        @JoinColumn(name = "user_id", referencedColumnName = "id")
        var user: User? = null
) : BaseTimeEntity()
